### ISSUE TITLE
error with translate.lua and idx_files

### ISSUE THREAD

**Issue by vince62s** [OpenNMT/OpenNMT#224]
I am getting this error

th translate.lua -model ../speech2textONMT/checkpoints/train_epoch5_207.04.t7 -src ../speech2textONMT/data/src-test.txt -output ../speech2textONMT/data/output-epoch5.txt -idx_files -gpuid 2[04/23/17 21:55:36 INFO] Using GPU(s): 2	
[04/23/17 21:55:36 WARNING] The caching CUDA memory allocator is enabled. This allocator improves performance at the cost of a higher GPU memory usage. To optimize for memory, consider disabling it by setting the environment variable: THC_CACHING_ALLOCATOR=0	
[04/23/17 21:55:36 INFO] Loading '../speech2textONMT/checkpoints/train_epoch5_207.04.t7'...	
[04/23/17 21:55:41 INFO] Model seq2seq trained on feattext	
/torch/install/bin/luajit: bad argument #2 to '?' (end index out of bound at /torch/pkg/torch/generic/Tensor.c:830)
stack traceback:
	[C]: at 0x7f9a4435b1c0
	[C]: in function '__newindex'
	./onmt/translate/DecoderAdvancer.lua:58: in function 'initBeam'
	./onmt/translate/BeamSearcher.lua:54: in function 'search'
	./onmt/translate/Translator.lua:273: in function 'translateBatch'
	./onmt/translate/Translator.lua:363: in function 'translate'
	translate.lua:110: in function 'main'
	translate.lua:196: in main chunk
	[C]: in function 'dofile'
	/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:150: in main chunk
	[C]: at 0x00406670

**Comment by vince62s** [comment]
@jsenellart-systran 
here: https://github.com/OpenNMT/OpenNMT/blob/master/onmt/translate/DecoderAdvancer.lua#L57
I fixed it with >0 but not 100% sure.

**Comment by shrutijpalaskar** [comment]
Hi, I got the same problem which got fixed with changing L57 of ./onmt/translate/DecoderAdvancer.lua as mentioned by @vince62s. Has there been any more insight into this problem?

**Comment by jsenellart-systran** [comment]
I will look at that today.

**Comment by jsenellart-systran** [comment]
Hello @shrutijpalaskar and @vince62s  - it seems that the issue is not related to input vector but to pbrnn - @shrutijpalaskar, do you confirm you are using it?
Since I don't have a model showing the problem, can you check that the problem setting batch size to 1 in translate.lua works? to double check, so that I can have a real fix. Thanks!

**Comment by shrutijpalaskar** [comment]
Hello @jsenellart-systran, yes I am using pdbrnn. I have tried using it with batch size 1 as well, but it gives the same error. Here are all the commands and the logs:

train - THC_CACHING_ALLOCATOR=0 th train.lua -data wsj/baseline_processed_feats-train.t7 -save_model WSJ_preprocessed_feats_2 -pdbrnn -report_every 1 -rnn_size 200 -word_vec_size 20 -layers 4 -max_batch_size 4 -learning_rate 0.7 -learning_rate_decay 0.8 -end_epoch 20 -gpuid 1 -log_file log/WSJ_processed_feats_gpu_6.out -save_every 100

translate - THC_CACHING_ALLOCATOR=0 th translate.lua -model WSJ_processed_feats_1_epoch1_37.81.t7 -src wsj/wsj_dev93.txt -batch_size 1 -gpuid 1 -idx_files

Error log - 
[05/17/17 09:22:34 INFO] Using GPU(s): 1        
[05/17/17 09:22:34 INFO] Loading 'WSJ_processed_feats_1_epoch1_37.81.t7'...     
[05/17/17 09:22:37 INFO] Model seq2seq trained on feattext   
/data/ASR1/spalaska/torch/install/bin/luajit: bad argument #2 to '?' (end index out of bound at /data/ASR1/spalaska/torch/pkg/torch/generic/Tensor.c:830)
stack traceback:
        [C]: at 0x7ffb3e0be1a0
        [C]: in function '__newindex'
        ./onmt/translate/DecoderAdvancer.lua:58: in function 'initBeam'
        ./onmt/translate/BeamSearcher.lua:54: in function 'search'
        ./onmt/translate/Translator.lua:273: in function 'translateBatch'
        ./onmt/translate/Translator.lua:363: in function 'translate'
        translate.lua:110: in function 'main'
        translate.lua:196: in main chunk
        [C]: in function 'dofile'
        ...aska/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:150: in main chunk
        [C]: at 0x00405d50

Thanks a lot for looking into this so quickly!

**Comment by ayushidalmia** [comment]
I am still getting this error while using translate.lua. However, it works fine for certain batch size which I randomly choose.

**Comment by jsenellart-systran** [comment]
Hi @ayushidalmia - did you retrain models? the code for pdbrnn and for  handling padding has been rewritten recently by @guillaumekln  - and you might need to retrain models. let us know if you can reproduce with recent code base.

**Comment by ayushidalmia** [comment]
My bad, I did not point out that I am using the default encoder. Further, I am training from scratch using the code checked out in the following commit:

https://github.com/OpenNMT/OpenNMT/commit/f05045cf66c2f344c1085d98cdbc0f8f5d9f9142
